BUILD

```diff
@@ -3341,6 +3341,266 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "tsi_ssl_types",
+    hdrs = [
+        "src/core/tsi/ssl_types.h",
+    ],
+    external_deps = [
+        "libssl",
+    ],
+    language = "c++",
+)
+
+grpc_cc_library(
+    name = "tsi_base",
+    srcs = [
+        "src/core/tsi/transport_security.cc",
```

Comment:
Is it going to cause problems that we have the same files listed in multiple targets here?

Could we avoid this by removing these files from their current targets and having those targets depend on these new targets?  For example, in the case of this `tsi_base` target, we could do the following:
- Remove `src/core/tsi/transport_security.cc`, `src/core/tsi/transport_security.h`, and `src/core/tsi/transport_security_interface.h` from the existing `tsi_interface` target, and have `tsi_interface` depend on `tsi_base`.
- Remove `src/core/tsi/transport_security_grpc.cc` and `src/core/tsi/transport_security_grpc.h` from the `tsi` target, and have `tsi` depend on `tsi_base`.

Similar things can be done for the other targets.

Reply:
Do you mean causing problems internally or in OSS? Internally (cl/407803147), we will replace the dependency of `grpc_secure` with `grpc_security_base`  and `tsi_interface` with `tsi_base`, respectively which means the internal targets will not depend on two targets that include the same files. WDYT?

Reply:
Sorry, let me take back my previous comment. It indeed results in a build failure internally due to the same file inclusion in multiple targets. Let me fix it as you suggested.